frontend/src/metabase-lib/join.ts

```diff
@@ -60,7 +83,15 @@ export function joinConditionOperators(
 export function suggestedJoinCondition(
   query: Query,
   stageIndex: number,
-  joinedThing: TableMetadata | CardMetadata,
-): ExternalOp | null {
-  return ML.suggested_join_condition(query, stageIndex, joinedThing);
+  joinable: Joinable,
+): FilterClause | null {
```

Comment:
After [discussion in Slack](https://metaboat.slack.com/archives/C04CYTEL9N2/p1686946743240819) I updated `suggested-join-condition` from #31590 to return an opaque MBQL clause rather than an external op